From 8b10fd9a2816b15ed11d49462cdeec99e95b419c Mon Sep 17 00:00:00 2001
From: Rajesh Kemisetti <rajeshk@codeaurora.org>
Date: Wed, 31 Oct 2012 12:19:32 +0530
Subject: [PATCH 09/75] msm: devices-msm7x27a: add new perf level for 8x25Q

Updating new turbo level based on the clock plan for 8x25Q.

Change-Id: I1116dd0dc3b4151b06a70fef765c517298217243
Signed-off-by: Rajesh Kemisetti <rajeshk@codeaurora.org>
---
 arch/arm/mach-msm/devices-msm7x27a.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-msm/devices-msm7x27a.c b/arch/arm/mach-msm/devices-msm7x27a.c
index 326c111..5eed49e 100644
--- a/arch/arm/mach-msm/devices-msm7x27a.c
+++ b/arch/arm/mach-msm/devices-msm7x27a.c
@@ -863,7 +863,11 @@ static struct resource kgsl_3d0_resources[] = {
 static struct kgsl_device_platform_data kgsl_3d0_pdata = {
 	.pwrlevel = {
 		{
-			.gpu_freq = 245760000,
+			.gpu_freq = 400000000,
+			.bus_freq = 200000000,
+		},
+		{
+			.gpu_freq = 320000000,
 			.bus_freq = 200000000,
 		},
 		{
@@ -876,7 +880,7 @@ static struct kgsl_device_platform_data kgsl_3d0_pdata = {
 		},
 	},
 	.init_level = 0,
-	.num_levels = 3,
+	.num_levels = 4,
 	.set_grp_async = set_grp_xbar_async,
 	.idle_timeout = HZ,
 	.strtstp_sleepwake = true,
@@ -913,9 +917,9 @@ void __init msm8x25_kgsl_3d0_init(void)
 
 		if (SOCINFO_VERSION_MAJOR(socinfo_get_version()) >= 2)
 			/* 8x25 v2.0 & above supports a higher GPU frequency */
-			kgsl_3d0_pdata.pwrlevel[0].gpu_freq = 320000000;
+			kgsl_3d0_pdata.pwrlevel[0].gpu_freq = 400000000;
 		else
-			kgsl_3d0_pdata.pwrlevel[0].gpu_freq = 300000000;
+			kgsl_3d0_pdata.pwrlevel[0].gpu_freq = 400000000;
 
 		kgsl_3d0_pdata.pwrlevel[0].bus_freq = 200000000;
 	}
-- 
1.7.0.4

